### PR TITLE
Revamped web3 transaction logic

### PR DESCRIPTION
**Pull request by kabilanvk**

### CODE REVIEW

File: src/Miningcore/Blockchain/Ethereum/EthereumPayoutHandler.cs [hunk 1]
```diff
@@ -81,6 +82,7 @@ public class EthereumPayoutHandler : PayoutHandlerBase, IPayoutHandler
         private const string TooManyTransactions = "There are too many transactions in the queue";
         private const decimal MaxPayout = 0.1m; // ethereum
         private const double MaxBlockReward = 0.1d; //ethereum
+        private static ConcurrentDictionary<string, string> transactionHashes = new();
```
Comment (kabilanvk) [inline src/Miningcore/Blockchain/Ethereum/EthereumPayoutHandler.cs:85]: I believe we will run master in a single node as of today. If we plan to run master in multiple nodes then this data should come from DB.

File: src/Miningcore/Blockchain/Ethereum/EthereumPayoutHandler.cs [hunk 2]
```diff
@@ -839,48 +841,95 @@ private async Task<double> GetNetworkBlockAverageTime(PoolConfig poolConfig)
...
+                        }
+                        // Get nonce for existing transaction
+                        var prevTx = await TelemetryUtil.TrackDependency(() => txService.Transactions.GetTransactionByHash.SendRequestAsync(prevTxHash),
+                            DependencyType.Web3, "GetTransactionByHash", $"addr={balance.Address},amt={balance.Amount.ToStr()},txhash={prevTxHash}");
+                        nonce = prevTx?.Nonce;
+                        logger.Info($"[{LogCategory}] Web3Tx receipt not found for existing tx. addr={balance.Address},amt={balance.Amount},txhash={prevTxHash},nonce={nonce}");
+                    }
+
+                    // Queue transaction
+                    txHash = await TelemetryUtil.TrackDependency(() => txService.GetEtherTransferService().TransferEtherAsync(balance.Address, balance.Amount, nonce: nonce),
```
Comment (joel-nlok) [inline src/Miningcore/Blockchain/Ethereum/EthereumPayoutHandler.cs:887]: we're getting the nonce from the previous transaction and then sending the next transaction with the same nonce?

don't we need to increment it?

Comment (kabilanvk) [reply src/Miningcore/Blockchain/Ethereum/EthereumPayoutHandler.cs:887]: We essentially resubmitting the transaction for the same address. Nonce won't change in that context according to documentation.

**Review by joel-nlok — APPROVED** [review]

File: src/Miningcore/Blockchain/Ethereum/EthereumPayoutHandler.cs (hunk 1 above)
Comment (joel-nlok) [reply src/Miningcore/Blockchain/Ethereum/EthereumPayoutHandler.cs:85]: yea.  splitting up the master responsibilities to multiple nodes is not easy given the current design.
even if we did, i suspect payouts would all be done by a single node,  balances by another, stats etc....  would be tricky to have multiple nodes doing payouts.

**Pull request merged**